### ISSUE TITLE
Transform AWS::LanguageExtensions seems not to break outputs according to conditions

### ISSUE THREAD

**Issue by jacky96623** [aws-cloudformation/cfn-language-discussion#125]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### **Tell us about the bug**
What goes wrong? Please describe the issue in as much detail as possible.
- I wrote a CloudFormation template that generally creates a vpc with necessary components (e.g., igw, subnets, route tables, S3 gateway VPC endpoint, etc.) accepting number of AZs to be spanned across. However, I discovered that the dynamically generated route tables are not attached to the VPC endpoint. Moreover, the output I defined (please refer to test case below for details) do not show the subnet IDs as well.

### **Expected behavior**
1. The dynamically generated route tables should be attached to the VPC endpoint according to conditions.
2. The output should show subnet IDs according to conditions.

### **Observed behavior**
1. The dynamically generated route tables are not attached to the VPC endpoint according to conditions.
2. The output does not show subnet IDs according to conditions.

### **Test cases**
Please include a minimal CloudFormation template here that reproduces the issue

template.yml
```yaml
AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::LanguageExtensions
Parameters:
  cidr:
    Type: String
    Default: 10.0.0.0/16
  noOfAZ:
    Type: Number
    MinValue: 1
    MaxValue: 2
    Default: 1
Conditions:
  noOfAZ1: !Equals
    - !Ref noOfAZ
    - 1
  noOfAZ2: !Equals
    - !Ref noOfAZ
    - 2
  azA: !Or
    - !Condition noOfAZ1
    - !Condition azB
  azB: !Condition noOfAZ2
Resources:
  vpc:
    Type: AWS::EC2::VPC
    Properties:
      CidrBlock: !Ref cidr
      EnableDnsHostnames: true
  publicSubnetA:
    Condition: azA
    Type: AWS::EC2::Subnet
    Properties:
      VpcId: !Ref vpc
      CidrBlock: !Select
        - 0
        - !Cidr
          - !Ref cidr
          - 16
          - 8
      AvailabilityZone: !Select
        - 0
        - Fn::GetAZs: !Ref AWS::Region
      MapPublicIpOnLaunch: true
  publicSubnetB:
    Condition: azB
    Type: AWS::EC2::Subnet
    Properties:
      VpcId: !Ref vpc
      CidrBlock: !Select
        - 1
        - !Cidr
          - !Ref cidr
          - 16
          - 8
      AvailabilityZone: !Select
        - 1
        - Fn::GetAZs: !Ref AWS::Region
      MapPublicIpOnLaunch: true

  privateSubnetA:
    Condition: azA
    Type: AWS::EC2::Subnet
    Properties:
      VpcId: !Ref vpc
      CidrBlock: !Select
        - 4
        - !Cidr
          - !Ref cidr
          - 16
          - 8
      AvailabilityZone: !Select
        - 0
        - Fn::GetAZs: !Ref AWS::Region
  privateSubnetB:
    Condition: azB
    Type: AWS::EC2::Subnet
    Properties:
      VpcId: !Ref vpc
      CidrBlock: !Select
        - 5
        - !Cidr
          - !Ref cidr
          - 16
          - 8
      AvailabilityZone: !Select
        - 1
        - Fn::GetAZs: !Ref AWS::Region

  igw:
    Type: AWS::EC2::InternetGateway
  igwAttachment:
    Type: AWS::EC2::VPCGatewayAttachment
    Properties:
      VpcId: !Ref vpc
      InternetGatewayId: !Ref igw
  publicRouteTable:
    Type: AWS::EC2::RouteTable
    Properties:
      VpcId: !Ref vpc
  publicRouteEgressRoute:
    Type: AWS::EC2::Route
    Properties:
      RouteTableId: !Ref publicRouteTable
      DestinationCidrBlock: 0.0.0.0/0
      GatewayId: !Ref igw

  privateRouteTableA:
    Condition: azA
    Type: AWS::EC2::RouteTable
    Properties:
      VpcId: !Ref vpc
  privateRouteTableB:
    Condition: azB
    Type: AWS::EC2::RouteTable
    Properties:
      VpcId: !Ref vpc

  publicSubnetARouteTableAssociation:
    Condition: azA
    Type: AWS::EC2::SubnetRouteTableAssociation
    Properties:
      RouteTableId: !Ref publicRouteTable
      SubnetId: !Ref publicSubnetA
  publicSubnetBRouteTableAssociation:
    Condition: azB
    Type: AWS::EC2::SubnetRouteTableAssociation
    Properties:
      RouteTableId: !Ref publicRouteTable
      SubnetId: !Ref publicSubnetB

  privateSubnetARouteTableAssociation:
    Condition: azA
    Type: AWS::EC2::SubnetRouteTableAssociation
    Properties:
      RouteTableId: !Ref privateRouteTableA
      SubnetId: !Ref privateSubnetA
  privateSubnetBRouteTableAssociation:
    Condition: azB
    Type: AWS::EC2::SubnetRouteTableAssociation
    Properties:
      RouteTableId: !Ref privateRouteTableB
      SubnetId: !Ref privateSubnetB

  s3GatewayVPCEndpoint:
    Type: AWS::EC2::VPCEndpoint
    Properties:
      ServiceName: !Sub com.amazonaws.${AWS::Region}.s3
      VpcEndpointType: Gateway
      VpcId: !Ref vpc
      RouteTableIds:
        - !Ref publicRouteTable
        - !If
          - azA
          - !Ref privateRouteTableA
          - !Ref AWS::NoValue
        - !If
          - azB
          - !Ref privateRouteTableB
          - !Ref AWS::NoValue

Outputs:
  publicSubnets:
    Value: !Join
      - ','
      - - !If
          - azA
          - !Ref publicSubnetA
          - !Ref AWS::NoValue
        - !If
          - azB
          - !Ref publicSubnetB
          - !Ref AWS::NoValue
  privateSubnets:
    Value: !Join
      - ','
      - - !If
          - azA
          - !Ref privateSubnetA
          - !Ref AWS::NoValue
        - !If
          - azB
          - !Ref privateSubnetB
          - !Ref AWS::NoValue
```

### **Additional context**
Anything else we should know?
Some remarks:
1. I use the transform to set DeletionPolicy dynamically but for the sake for minimization removed those lines.
2. If you remove the transform line and re-create the stack, the whole thing works correctly.

Some relevant screenshots:
Regarding template WITH transform:
![image](https://user-images.githubusercontent.com/37379550/235824701-d7673f36-cf2a-410e-a5a7-674663275e93.png)
![image](https://user-images.githubusercontent.com/37379550/235824751-aed3adb4-c9c8-4a90-a9a8-e1afb80bd56d.png)

Regarding template WITHOUT transform:
![image](https://user-images.githubusercontent.com/37379550/235824794-1b457119-a6b0-41b3-88cb-9f27024b6639.png)
![image](https://user-images.githubusercontent.com/37379550/235824856-ec86adeb-5674-444d-80c7-602eb5bf00d6.png)

**Comment by MalikAtalla-AWS** [comment]
The issue with the output values is due to a bug in how we resolve default values of parameters of type Number. I'm currently working on a fix.

**Comment by MalikAtalla-AWS** [comment]
The bugfix is merged into our code base and will start to make its way to production regions. We will update this issue once the rollout is complete.

**Comment by MalikAtalla-AWS** [comment]
The bugfix is rolled out to all regions, except GovCloud. GovCloud deployment will happen tomorrow.
